echopype/echodata/zarr_combine.py

```diff
@@ -916,14 +917,14 @@ def _modify_prov_filenames(
 
         # obtain the filenames zarr array
         zarr_filenames = zarr.open_array(
-            zarr_path + "/Provenance/filenames", mode="r+", storage_options=storage_options
+            zarr_path / "Provenance/filenames", mode="r+", storage_options=storage_options
```

Comment:
Seems like you're forcing `zarr_path` to be pathlib.Path only? How about if I want to pass in `s3://my-bucket/data.zarr`? casting that to `Path` will make that not work, turning the string to `s3:/my-bucket/data.zarr`.